### PR TITLE
Add NodeToNodeVersionData CDDL tests

### LINKED ISSUE #4540: Write cddl based tests for NodeTo{Client,Node}VersionData
These are not tested with `prop_encodeHandshakeNodeTo{Client,Node}...`
functions as the handshake codec only decodes the
`NodeTo{Client,Node}VersionData` to cbor `Term`.

### PR DESCRIPTION
**Pull request by bolt12**
# Description

Closes #4540

### CODE REVIEW

**Comment by coot** [comment]
It seems there are `stylish-haskell` issues.

File: ouroboros-network-protocols/test-cddl/Main.hs [hunk 1]
```diff
@@ -451,7 +498,7 @@ instance Arbitrary NtNHandshakeV7To10 where
 
 instance Arbitrary NtNHandshakeV11ToLast where
   arbitrary = do
-    let genVersion = elements [NodeToNodeV_12 ..]
+    let genVersion = elements [NodeToNodeV_11 ..]
```
Comment (coot) [inline ouroboros-network-protocols/test-cddl/Main.hs:501]: Good catch!

File: ouroboros-network-protocols/test-cddl/Main.hs [hunk 2]
```diff
@@ -590,6 +637,61 @@ prop_encodePeerSharing
...
+
+newtype NtNVersionDataV11ToLast = NtNVersionDataV11ToLast (NodeToNodeVersion, NodeToNodeVersionData)
+  deriving Show
+
+instance Arbitrary NtNVersionDataV11ToLast where
+  arbitrary = do
+    NtNVersionV11ToLast ntnVersion <- arbitrary
+    ntnVersionData <- arbitrary
+    let ntnVersionData' = case ntnVersion of
+          NodeToNodeV_11 -> ntnVersionData { NtNVersion.query = False }
```
Comment (coot) [inline ouroboros-network-protocols/test-cddl/Main.hs:674]: Why do we set the query flag to `False` in `NodeToNodeV_11`?  It was added in `NodeToNodeV_11`.

File: ouroboros-network-protocols/test-cddl/specs/node-to-node-version-data-v11.cddl [hunk 3]
```diff
@@ -0,0 +1,12 @@
+;
+; NodeToNodeVersionData, v12
```
Comment (coot) [inline ouroboros-network-protocols/test-cddl/specs/node-to-node-version-data-v11.cddl:2]:
```suggestion
; NodeToNodeVersionData, v11
```

**Review by coot — CHANGES_REQUESTED** [review]

File: ouroboros-network-protocols/test-cddl/Main.hs (hunk 2 above)
Comment (bolt12) [reply ouroboros-network-protocols/test-cddl/Main.hs:674]: I thought it got added in V_12, V_11 was PeerSharing, right?

Comment (bolt12) [reply ouroboros-network-protocols/test-cddl/Main.hs:674]: nvm, you are right

**Review by coot — APPROVED** [review]
LGTM

**Pull request merged**